File: java/client/src/main/java/glide/api/models/commands/geospatial/GeoUnit.java
/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
package glide.api.models.commands.geospatial;

import glide.api.commands.GeospatialIndicesBaseCommands;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

/**
 * Enumeration representing distance units options for the {@link
 * GeospatialIndicesBaseCommands#geodist(String, String, String, GeoUnit)} command.
 */
@RequiredArgsConstructor
@Getter
public enum GeoUnit {
    /** Represents distance in meters. */
    METERS("m"),
    /** Represents distance in kilometers. */
    KILOMETERS("km"),
    /** Represents distance in miles. */
    MILES("mi"),
    /** Represents distance in feet. */
    FEET("ft");

    private final String redisApi;
}

File: csharp/sources/Valkey.Glide/ser_Enums/GeoUnit.cs
using System;

namespace Valkey.Glide;

/// <summary>
/// Units associated with Geo Commands.
/// </summary>
public enum GeoUnit
{
    /// <summary>
    /// Meters.
    /// </summary>
    Meters,

    /// <summary>
    /// Kilometers.
    /// </summary>
    Kilometers,

    /// <summary>
    /// Miles.
    /// </summary>
    Miles,

    /// <summary>
    /// Feet.
    /// </summary>
    Feet,
}

internal static class GeoUnitExtensions
{
    internal static ValkeyValue ToLiteral(this GeoUnit unit) => unit switch
    {
        GeoUnit.Feet => ValkeyLiterals.ft,
        GeoUnit.Kilometers => ValkeyLiterals.km,
        GeoUnit.Meters => ValkeyLiterals.m,
        GeoUnit.Miles => ValkeyLiterals.mi,
        _ => throw new ArgumentOutOfRangeException(nameof(unit)),
    };
}
